cmake/Findh5cpp.cmake

```diff
@@ -0,0 +1,7 @@
+find_path(H5CPP_INCLUDE_DIR NAMES h5cpp/hdf5.hpp)
```

Comment:
Just curious as to why this file is needed? I thought that the `h5cpp-config.cmake`, which is added to `/usr/local/lib/cmake` should be capable of finding it if you installed h5cpp to system.
As I understand, most of the changes below are for the sake of the older centos sytems?

Reply:
The rationale was to be explicit about finding the headers, also in the case when `h5cpp` is not installed in a system location.

> As I understand, most of the changes below are for the sake of the older centos sytems?

If you refer to adding the header directory to `path_include_common` and the libraries to `libraries_common` then this is to support both Conan as well as non-Conan builds in a, at least in my subjective and certainly biased opinion, minimal and non-magic way.